api/pipeline/compile.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2022 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+// nolint: dupl // ignore similar code with expand
+package pipeline
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-vela/server/compiler"
+	"github.com/go-vela/server/router/middleware/org"
+	"github.com/go-vela/server/router/middleware/pipeline"
+	"github.com/go-vela/server/router/middleware/repo"
+	"github.com/go-vela/server/router/middleware/user"
+	"github.com/go-vela/server/util"
+	"github.com/go-vela/types"
+	"github.com/sirupsen/logrus"
+)
+
+// swagger:operation POST /api/v1/pipelines/{org}/{repo}/{pipeline}/compile pipelines CompilePipeline
+//
+// Get, expand and compile a pipeline from the configured backend
+//
+// ---
+// produces:
+// - application/x-yaml
+// - application/json
+// parameters:
+// - in: path
+//   name: repo
+//   description: Name of the repo
+//   required: true
+//   type: string
+// - in: path
+//   name: org
+//   description: Name of the org
+//   required: true
+//   type: string
+// - in: path
+//   name: pipeline
+//   description: Commit SHA for pipeline to retrieve
+//   required: true
+//   type: string
+// - in: query
+//   name: output
+//   description: Output string for specifying output format
+//   type: string
+// security:
+//   - ApiKeyAuth: []
+// responses:
+//   '200':
+//     description: Successfully retrieved and compiled the pipeline
+//     schema:
+//       "$ref": "#/definitions/PipelineBuild"
+//   '400':
+//     description: Unable to validate the pipeline configuration
+//     schema:
+//       "$ref": "#/definitions/Error"
+//   '404':
+//     description: Unable to retrieve the pipeline configuration
+//     schema:
+//       "$ref": "#/definitions/Error"
+
+// CompilePipeline represents the API handler to capture,
+// expand and compile a pipeline configuration.
+func CompilePipeline(c *gin.Context) {
+	// capture middleware values
+	m := c.MustGet("metadata").(*types.Metadata)
+	o := org.Retrieve(c)
+	p := pipeline.Retrieve(c)
+	r := repo.Retrieve(c)
+	u := user.Retrieve(c)
+
+	entry := fmt.Sprintf("%s/%s", r.GetFullName(), p.GetCommit())
+
+	// update engine logger with API metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Entry.WithFields
+	logrus.WithFields(logrus.Fields{
+		"org":      o,
+		"pipeline": p.GetCommit(),
+		"repo":     r.GetName(),
+		"user":     u.GetName(),
+	}).Infof("compiling pipeline %s", entry)
```

Comment:
We certainly could 👍 I'm leaning towards leaving it in there since it doesn't hurt.

This pattern is adopted from existing log entires we already established for other API endpoints.

So if we did want to switch this up, I think we'd have to go back and apply the same changes for consistency.

Reply:
Fair point, I'd vote for removing it maybe in another PR. Seems like a waste of bytes that can add up - then again, this is more compact than logging as separate JSON fields :disappear:

Reply:
Yeah I see where you're coming from on that 👍 

To me, the message for the log entry is designed for human processing

i.e. user who manually reviews the log entries or using a tool such as `grep` or `awk`

In those scenarios, it can add a bit of complexity when trying to search for multiple fields but not impossible:

```shell
# searching for specific message
cat log.txt | grep '<org>/<repo>/<commit>'

# searching for specific fields
cat log.txt | grep '"org": <org>' | grep '"repo": <repo>' | grep '"pipeline": <commit>'
```

While the fields for the log entry are designed for system processing

i.e. most apps/tools consuming logs enable filtering on one or more fields which improves this experience

However, if folks feel strongly enough about it, then I could get behind shortening the message